### PR TITLE
change owner error removed if only one user exist in the group/channel

### PR DESCRIPTION
**Pull request by AnshD12**
change owner error removed if only one user exist in the group/channel 

BUG

@RocketChat/core 

previously if a user wanted to leave a group/channel there used to be an error related to change owner (screenshot 1) which is not required if he is the only member oh that group/channel.

image

I have changed leaveRoom.js file to avoid that error now

### CODE REVIEW

**Comment by AnshD12** [comment]
@geekgonecrazy did you checked this ??

**Comment by geekgonecrazy** [comment]
If your the last user we probably shouldn't just let them leave.  We should probably delete it.  Otherwise the room is memberless

**Comment by AnshD12** [comment]
@geekgonecrazy done that

File: packages/rocketchat-lib/server/methods/leaveRoom.js [hunk 1]
```diff
@@ -24,7 +24,13 @@ Meteor.methods({
 		if (RocketChat.authz.hasRole(user._id, 'owner', room._id)) {
 			const numOwners = RocketChat.authz.getUsersInRole('owner', room._id).fetch().length;
 			if (numOwners === 1) {
-				throw new Meteor.Error('error-you-are-last-owner', 'You are the last owner. Please set new owner before leaving the room.', { method: 'leaveRoom' });
+				if (room.usernames.length > 1) {
+					throw new Meteor.Error('error-you-are-last-owner', 'You are the last owner. Please set new owner before leaving the room.', { method: 'leaveRoom' });
+				} else {
+					RocketChat.models.Messages.removeByRoomId(rid);
```
Comment (vynmera) [inline packages/rocketchat-lib/server/methods/leaveRoom.js]: We should warn the user that leaving an empty room will destroy the room, together with its history. Leaving a room usually doesn't imply "delete its history", so this change in action could lead to confusion and data loss.

**Comment by pierre-lehnen-rc** [comment]
This issue is no longer relevant in the current version of Rocket.Chat, as the behavior was changed.

**Pull request closed without merging**